elf_load.c

```diff
@@ -311,43 +312,17 @@ static int elf_load_imgcpy(struct elf_prog *elf_prog, Elf *elf,
 }
 
 #if CONFIG_LIBVFSCORE
-/* Read from fd exact `len` bytes from offset `roff`, fail otherwise */
-static int elf_load_do_fdread(int fd, off_t roff, void *dst, size_t len)
```

Comment:
@mogasergiu: Can we keep `elf_load_do_fdread()` for builds without `mmap` support (e.g., `#if !CONFIG_LIBPOSIX_MMAP`)? I'd introduce a new function set with `_fdmap()` suffix.